### PR TITLE
Modify Coherence to use v3 NuGet APIs

### PR DESCRIPTION
**Pull request by pranavkm**

### CODE REVIEW

**Comment by pranavkm** [comment]
cc @davidfowl \ @victorhurdugaci

**Comment by davidfowl** [comment]
@emgarten for sanity check on API usage

File: src/CoherenceBuild/CoherenceVerifier.cs [hunk 1]
```diff
@@ -32,19 +29,10 @@ public static bool VerifyAll(ProcessResult result)
...
-                        foreach (var mismatch in packageInfo.DependencyMismatches)
-                        {
-                            Log.WriteError("    Expected {0}({1}) but got {2}",
-                                mismatch.Dependency,
-                                (mismatch.TargetFramework == VersionUtility.UnsupportedFrameworkName ?
-                                "NETSTANDARDAPP1_5" :
-                                VersionUtility.GetShortFrameworkName(mismatch.TargetFramework)),
-                                mismatch.Info.Package.Version);
-                        }
+                        Log.WriteError($"{packageInfo.Identity} depends on {mismatch.Dependency.Id} " +
```
Comment (davidfowl) [inline src/CoherenceBuild/CoherenceVerifier.cs]: Where's the target framework information?

File: src/CoherenceBuild/PackagePublisher.cs [hunk 2]
```diff
@@ -54,39 +56,40 @@ public static void PublishToFeed(ProcessResult processResult, string feed, strin
 
         private static async Task PublishToFeedAsync(ProcessResult processResult, string feed, string apiKey)
         {
-            var server = new PackageServer(feed, "Custom DNX");
```
Comment (davidfowl) [inline src/CoherenceBuild/PackagePublisher.cs]: What's the new user agent?

File: src/CoherenceBuild/CoherenceVerifier.cs (hunk 1 above)
Comment (pranavkm) [inline src/CoherenceBuild/CoherenceVerifier.cs]: I'll that back.

File: src/CoherenceBuild/PackagePublisher.cs (hunk 2 above)
Comment (pranavkm) [inline src/CoherenceBuild/PackagePublisher.cs]: No idea where that is now. I guess it's just using NuGet defaults

**Pull request closed without merging**